src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -11575,6 +11575,21 @@ BYTE* emitter::emitOutputRRR(BYTE* dst, instrDesc* id)
 
     noway_assert(!id->idGCref());
 
+    if (!emitInsCanOnlyWriteSSE2OrAVXReg(id))
```

Comment:
Odd that it checks the instruction instead of just checking if reg1 is XMM or GPR.

Reply:
That seems to be the prevailing pattern.

If the `IF_xxx` were fully descriptive, seems like we could sink all these register write GC updates into the tail end of some higher-level emit method and rely on the form to tell us which registers were written... then we'd only need to special case calls and such.